src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -127,6 +162,79 @@ def getTaskData(environment_name, module_name, task_name)
 
   private
 
+  # @return Puppet::Node::Facts facts, Hash trusted_facts
+  def process_facts(request_data)
+    if request_data['facts'].nil?
+      facts = get_facts_from_pdb(request_data['certname'], request_data['environment'])
+    else
+      facts_from_request = request_data['facts']
+      facts_from_request['name'] = request_data['certname']
+      facts = Puppet::Node::Facts.from_data_hash(facts_from_request)
+    end
+
+    fact_values = if facts.nil?
+                    {}
+                  else
+                    facts.sanitize
```

Comment:
Feels weird to me to call sanitize here - like this conditional is turning whatever we have into a data structure, not doing additional, conditional processing.

Reply:
I don't know, does putting the conditional sanitization up in the above conditional or something like this make sense?
```rb
facts.sanitize if facts
fact_values = facts ? facts.to_data_hash['values'] : {}
```

Reply:
Yeah I had just written something like the second one. I think it's clearer. Because the PDB terminus can explicitly return `nil` sometimes: https://github.com/puppetlabs/puppetdb/blob/master/puppet/lib/puppet/indirector/facts/puppetdb.rb#L84.

Reply:
**Oh**! I see.... We only use this for finding `trusted_facts` _if_ `request_data['trusted_facts']` is `nil`.

Reply:
_And_ `Node#merge` calls `#sanitize` on the facts that are passed into it, which in our case is the `facts` variable.

Reply:
Yes. I still think splitting out sanitize makes sense though. The new thing is much shorter and clearer.

Reply:
What do you think about putting the creation of `fact_values` inside the `else` branch of the `trusted_facts` conditional? Does that make your above point more obvious?